dango/dex/src/execute.rs

```diff
@@ -162,8 +163,25 @@ fn cancel_orders(ctx: MutableCtx, order_ids: BTreeSet<OrderId>) -> anyhow::Resul
     let mut events = Vec::new();
 
     for order_id in order_ids {
-        let (((base_denom, quote_denom), direction, price, _), order) =
-            ORDERS.idx.order_id.load(ctx.storage, order_id)?;
+        // First try to load from resting orders, then from incoming orders storage.
+        // If found in neither or both through an error
+        let (((base_denom, quote_denom), direction, price, _), order) = match (
+            ORDERS.idx.order_id.may_load(ctx.storage, order_id)?,
+            INCOMING_ORDERS.may_load(ctx.storage, order_id)?,
+        ) {
+            (Some(x), None) => {
+                ORDERS.remove(ctx.storage, x.0.clone())?;
+                x
+            },
```

Comment:
@pacmanifold Don't name variables like `x`. It's hard to read. I also really hate syntax like `x.0`.

Do something like this instead:

```rust
(Some((order_key, order)), None) => {
    ORDERS.remove(ctx.storage, order_key.clone())?;
    (order_key, order)
}
```